src/components/wallet/methods/updateAccount.ts

```diff
@@ -4,13 +4,47 @@ import {
   // map as loMap
 } from 'lodash-es'
 
+import { Keypair, Networks } from 'stellar-sdk'
 import { handleError } from '@services/error'
+import { set, get } from '@services/storage'
 import { Wallet } from '../wallet'
 
-export default async function updateAccount(this: Wallet) {
+export default async function updateAccount(
+  this: Wallet,
+  { displayPrompt }: { displayPrompt?: boolean } = {}
+) {
   try {
     this.error = null
     this.loading = { ...this.loading, update: true }
+    let keypair
+
+    if (displayPrompt || !this.account.secretKey) {
+      let inputs = await this.setPrompt({
+        message: 'Enter your Stellar secret key',
+      })
+      let secret = inputs[0].value
+      try {
+        keypair = Keypair.fromSecret(secret)
+      } catch (e) {
+        throw 'Invalid secret key'
+      }
+      this.account = {
+        ...this.account,
+        publicKey: keypair.publicKey(),
+        secretKey: keypair.secret(),
+      }
+      let query = `?secretKey=${this.account.secretKey}`
+      if (this.network_passphrase === Networks.PUBLIC) {
+        query += `&pubnet=true`
+      }
+      history.replaceState(null, '', query)
+      set('WALLET[keystore]', btoa(JSON.stringify(this.account)))
+      set(
+        'UNTRUSTEDASSETS[keystore]',
+        btoa(JSON.stringify(this.UntrustedAssets))
+      )
```

Comment:
aha! So it does work haha. Yea just use `this.UntrustedAssets` every time instead of building the `UNTRUSTEDASSETS` map.

Reply:
Unfortunately I can't avoid build the map each time because for whatever reason when javascript attempts to encode the objects attribute directly the result when decoded is an empty map; even after we have populated the map.

```
!!!ENCODING DIRECTLY!!!

set('UNTRUSTEDASSETS[keystore]', btoa(JSON.stringify(this.UntrustedAssets))) 
=  
e30=

!!!DECODING THE DIRECT ATTRIBUTE!!!

const TEST = await get('UNTRUSTEDASSETS[keystore]')
const mapTEST = new Map(
        Object.entries(JSON.parse(atob(TEST)))
        )
=
Map(0) {}[[Entries]]No propertiessize: (...)__proto__: Map
------------------------------------------------------------------
!!!ENCODING THE BUILT MAP!!!

let UNTRUSTEDASSETS = {}
      this.UntrustedAssets.forEach((value, key) => {
        UNTRUSTEDASSETS[key] = value
      })
set('UNTRUSTEDASSETS[keystore]', btoa(JSON.stringify(UNTRUSTEDASSETS)))
 =  eyJTUlQ6R0NETkpVQlFTWDdBSldMSkFDTUo3STRCQzNaNDdCUVVUTUhFSUNaTEU2TVU0S1FCUllHNUpZNkIiOnsiYXNzZXRfY29kZSI6IlNSVCIsImFzc2V0X2lzc3VlciI6IkdDRE5KVUJRU1g3QUpXTEpBQ01KN0k0QkMzWjQ3QlFVVE1IRUlDWkxFNk1VNEtRQlJZRzVKWTZCIiwiYmFsYW5jZSI6IjAuMDAwMDAwMCIsImFzc2V0X3R5cGUiOiJjcmVkaXRfYWxwaGFudW00IiwiaXNfYXV0aG9yaXplZCI6dHJ1ZSwidW50cnVzdGVkIjp0cnVlfX0=

!!!DECODING THE BUILT MAP!!!

const TEST2UNTRUSTEDASSETS = await get('UNTRUSTEDASSETS[keystore]')
const map2TESTUNTRUST = new Map(
        Object.entries(JSON.parse(atob(TEST2UNTRUSTEDASSETS)))
)
= Map(1) {"SRT:GCDNJUBQSX7AJWLJACMJ7I4BC3Z47BQUTMHEICZLE6MU4KQBRYG5JY6B" => {…}}[[Entries]]0: {"SRT:GCDNJUBQSX7AJWLJACMJ7I4BC3Z47BQUTMHEICZLE6MU4KQBRYG5JY6B" => Object}size: (...)__proto__: Map
```

Reply:
I looked around but I didnt find another way to stringify the object's map attribute without decoupling it from object it self (ie. building it).

Frankly idk why this  works at all -> `set('WALLET[keystore]', btoa(JSON.stringify(this.account)))`

Maybe it doesnt, I would have to investigate further to see if it does or not (the system behaves as if it does...).

Reply:
If you have any suggestions Im open to hearing them because it was a weird bug.

Reply:
Technically to be more "correct" I should change this to 
```suggestion
      let UNTRUSTEDASSETS = {}
      this.UntrustedAssets.forEach((value, key) => {
         UNTRUSTEDASSETS[key] = value
      })
     set('UNTRUSTEDASSETS[keystore]', btoa(JSON.stringify(UNTRUSTEDASSETS)))
```
However since `set('UNTRUSTEDASSETS[keystore]', btoa(JSON.stringify(UNTRUSTEDASSETS)))` already makes an empty map in a stage where an empty map is required it could stay as is.

Reply:
I'd like to figure this out, because I don't believe there's some innate problem with serializing `this.UntrustedAssets`. I tested this in a JS console and there doesn't seem to be any difference in serialization between a `Map()` and an `object`:
![Screen Shot 2021-01-14 at 10 04 30 AM](https://user-images.githubusercontent.com/10968980/104630794-1daa3b00-5650-11eb-922d-25bbc0be6b15.png)

Reply:
Is it possible that `UNTRUSTEDASSETS[keystore]` is being overwritten and thats why it's empty?

Reply:
I don't think I'm following.
I changed the objects attribute to include the new asset.

```
this.UntrustedAssets.set(
        `${addAsset.asset_code}:${addAsset.asset_issuer}`,
        {
          asset_code: addAsset.asset_code,
          asset_issuer: addAsset.asset_issuer,
          balance: '0.0000000',
          asset_type: addAsset.asset_type,
          is_authorized: true,
          untrusted: true,
        }
      )
```
Then I attempt to overwrite `UNTRUSTEDASSETS[keystore]` with the new changes.
```
set(
        'UNTRUSTEDASSETS[keystore]',
        btoa(JSON.stringify(this.UntrustedAssets))
      )
```
However it doesnt serialize it properly. only giving me an empty map.

Reply:
Perhaps its the way I'm setting it.
I could try the key value approach instead of `set`
```
this.UntrustedAssets[`${addAsset.asset_code}:${addAsset.asset_issuer}`] = {
          asset_code: addAsset.asset_code,
          asset_issuer: addAsset.asset_issuer,
          balance: '0.0000000',
          asset_type: addAsset.asset_type,
          is_authorized: true,
          untrusted: true,
        }
```
Worth a shot.